scripts/install.sh

```diff
@@ -78,22 +91,81 @@ if [ "$NETWORK" == "development" ]; then
         npx hardhat unlock-accounts --network $NETWORK
 fi
 
-printf "${LOG_START}Building contracts...${LOG_END}"
-yarn clean
-yarn build
+# deploy beacon
+printf "${LOG_START}Building beacon...${LOG_END}"
+yarn clean && yarn build
 
-if [ "$SKIP_DEPLOYMENT" = false ] ; then
-  printf "${LOG_START}Deploying contracts...${LOG_END}"
+printf "${LOG_START}Deploying contracts for beacon...${LOG_END}"
+USE_EXTERNAL_DEPLOY=true npx hardhat deploy --reset --export export.json --network $NETWORK
 
+# create link to random-beacon
+yarn link
+
+if [ "$SKIP_ECDSA_DEPLOYMENT" = false ] && [ "$SKIP_TBTC_DEPLOYMENT" = false ]; then
+  cd $KEEP_ECDSA_SOL_PATH
+
+  printf "${LOG_START}Building ecdsa...${LOG_END}"
+  yarn && yarn clean && yarn build
+
+  # link random-beacon
+  yarn link @keep-network/random-beacon
+
+  # deploy ecdsa
+  printf "${LOG_START}Deploying ecdsa contracts...${LOG_END}"
   npx hardhat deploy --reset --export export.json --network $NETWORK
-fi
+  
+  # create a link to ecdsa
+  yarn link
+
+  # create export folder
+  yarn prepack
+
+  cd $KEEP_CORE_PATH
+
+  if [ "$TBTC_PATH" = "" ]; then
+    printf "${LOG_START}Cloning tbtc...${LOG_END}"
+    # create a temporary tbtc dir for fresh installation
+    rm -rf $TMP_TBTC && mkdir $TMP_TBTC && cd $TMP_TBTC
+    # clone project from the repository
+    git clone https://github.com/keep-network/tbtc-v2.git
     
-if [ "$SKIP_CLIENT_BUILD" = false ] ; then
-   printf "${LOG_START}Building keep-core client...${LOG_END}"
+    printf "${LOG_START}Building tbtc contracts...${LOG_END}"
+    cd "tbtc-v2/solidity"
+    yarn install && yarn build
+  else
+    printf "${LOG_START}Installing tbtc from the local directory...${LOG_END}"
+    cd "$TBTC_PATH/solidity"
+  fi
+
+  # link random-beacon contracts
+  yarn link @keep-network/random-beacon
+  # link ecdsa contracts
+  yarn link @keep-network/ecdsa
+
+  # deploy tbtc
+  printf "${LOG_START}Deploying tbtc contracts...${LOG_END}"
+  npx hardhat deploy --reset --export export.json --network $NETWORK
+
+elif [ "$SKIP_ECDSA_DEPLOYMENT" = false ] && [ "$SKIP_TBTC_DEPLOYMENT" = true ]; then
+  # deploy ecdsa
+  cd $KEEP_ECDSA_SOL_PATH
+
+  printf "${LOG_START}Building ecdsa...${LOG_END}"
+  yarn && yarn clean && yarn build
+
+  # link random-beacon
+  yarn link @keep-network/random-beacon
+
+  printf "${LOG_START}Deploying ecdsa contracts...${LOG_END}"
+  npx hardhat deploy --reset --export export.json --network $NETWORK
+fi
+
+if [ "$SKIP_CLIENT_BUILD" = false ]; then
+   printf "${LOG_START}Building client...${LOG_END}"
 
    cd $KEEP_CORE_PATH
    go generate ./...
    go build -a -o keep-core .
```

Comment:
Maybe we could use the `make` command to use the `Makefile` that was added in the root directory?
```suggestion
   make
```

Reply:
done